webpack.config.js

```diff
@@ -108,24 +108,54 @@ const developmentConfig = {
 
 // special config for production
 const productionConfig = {
+  bail: true, // Stop the build on errors
   plugins: [
     new webpack.DefinePlugin({
       'process.env.NODE_ENV': JSON.stringify('production')
     })
   ],
   optimization: {
     minimize: true,
-    minimizer: [new TerserPlugin()]
-  }
+    minimizer: [new TerserPlugin()],
+  },
+}
+
+// additional config for testing
+const testingConfig = {
+  bail: true, // Stop on errors or critical warnings
+  performance: {
+    hints: false // Suppress performance warnings in testing
+  },
+  ignoreWarnings: [
+    { message: /performance recommendations/ },
+    { message: /asset size limit/ },
+    { message: /entrypoint size limit/ }
+  ]
 }
 
 // combine config depending on the provided --mode command line option
 module.exports = (env, argv) => {
+  // Determine if testing mode is enabled
+  const isTesting = env && env.testing
+
   return configList.map(config => {
     if (argv.mode === 'development') {
+      // Return the development configuration directly
       return merge(config, baseConfig, developmentConfig)
+    } else if (argv.mode === 'production') {
+      // Start with the base production configuration
+      let finalConfig = merge(config, baseConfig, productionConfig)
+
+      // Add testing-specific configuration if in testing mode
+      if (isTesting) {
+        finalConfig = merge(finalConfig, testingConfig)
+      }
+
+      return finalConfig
     } else {
-      return merge(config, baseConfig, productionConfig)
+      throw new Error(
```

Comment:
To much complexity `production` can be the fallback and should be very strict, but still needs to allow for larger files.

Reply:
yes, refactored it into a `switch` `case`, I don't see the need for a fallback when it should not be receiving any other mode..